app/src/main/java/io/github/sds100/keymapper/actions/pinchscreen/PichPickDisplayCoordinateFragment.kt

```diff
@@ -0,0 +1,153 @@
+import android.provider.MediaStore
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import android.widget.ArrayAdapter
+import androidx.activity.addCallback
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.getSystemService
+import androidx.core.graphics.decodeBitmap
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.setFragmentResult
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.*
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import io.github.sds100.keymapper.databinding.FragmentPinchPickCoordinatesBinding
+import io.github.sds100.keymapper.system.files.FileUtils
+import io.github.sds100.keymapper.util.*
+import io.github.sds100.keymapper.util.ui.showPopups
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+class PinchPickDisplayCoordinateFragment : Fragment() {
+    companion object {
+        const val EXTRA_RESULT = "extra_result"
+    }
+
+    private val args: PinchPickDisplayCoordinateFragmentArgs by navArgs()
+    private val requestKey: String by lazy { args.requestKey }
+    private var pinchTypesDisplayValues = mutableListOf<String>()
+
+    private val viewModel: PinchPickDisplayCoordinateViewModel by viewModels {
+        Inject.pinchCoordinateActionTypeViewModel(requireContext())
+    }
+
+    private val screenshotLauncher =
+        registerForActivityResult(ActivityResultContracts.GetContent()) {
+            it ?: return@registerForActivityResult
+
+            val bitmap = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                ImageDecoder.createSource(requireContext().contentResolver, it)
+                    .decodeBitmap { _, _ -> }
+            } else {
+                MediaStore.Images.Media.getBitmap(requireContext().contentResolver, it)
+            }
+
+            val displaySize = Point().apply {
+                val windowManager: WindowManager = requireContext().getSystemService()!!
+                windowManager.defaultDisplay.getRealSize(this)
+            }
+
+            viewModel.selectedScreenshot(bitmap, displaySize)
+        }
+
+    /**
+     * Scoped to the lifecycle of the fragment's view (between onCreateView and onDestroyView)
+     */
+    private var _binding: FragmentPinchPickCoordinatesBinding? = null
+    val binding: FragmentPinchPickCoordinatesBinding
+        get() = _binding!!
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        pinchTypesDisplayValues = arrayOf(getString(io.github.sds100.keymapper.R.string.hint_coordinate_type_PINCH_IN), getString(io.github.sds100.keymapper.R.string.hint_coordinate_type_PINCH_OUT)).toMutableList();
```

Comment:
You should be able to import the `R` in `R.string.hint_coordinate_type_PINCH_IN` without using the absolute package name.

Reply:
Sure, that was a strange thing in Android Studio ... it won't let me use it until I manually added the proper import for that ...